app/src/main/java/com/github/se/eventradar/ui/event/EventDetails.kt

```diff
@@ -18,159 +16,43 @@ import androidx.compose.material3.MaterialTheme
-fun EventDescription(modifier: Modifier, contentColor: Color, titleColor: Color) {
-  Column(modifier = modifier) {
-    Text(
-        text = headerDescription,
-        style = subTitleTextStyle,
-        color = titleColor,
-        modifier = Modifier.testTag("descriptionTitle"))
-    Text(
-        text = contentDescription,
-        style = standardTextStyle,
-        color = contentColor,
-        modifier = Modifier.testTag("descriptionContent"))
-  }
-}
-
-@Composable
-fun EventDistance(modifier: Modifier, contentColor: Color, titleColor: Color) {
-  Column(modifier = modifier) {
-    Text(
-        text = headerDistance,
-        style = subTitleTextStyle,
-        color = titleColor,
-        modifier = Modifier.testTag("distanceTitle"))
-    Text(
-        text = contentDistance,
-        style = standardTextStyle,
-        color = contentColor,
-        modifier = Modifier.testTag("distanceContent"))
-  }
-}
 
 @Composable
-fun EventDate(modifier: Modifier, contentColor: Color, titleColor: Color) {
-  Column(modifier = modifier) {
-    Text(
-        text = headerDate,
-        style = subTitleTextStyle,
-        color = titleColor,
-        modifier = Modifier.testTag("dateTitle"))
-    Text(
-        text = contentDate,
-        style = standardTextStyle,
-        color = contentColor,
-        modifier = Modifier.testTag("dateContent"))
-  }
-}
+fun EventDetails(
+    viewModel: EventDetailsViewModel = viewModel(),
+    navigationActions: NavigationActions
+) {
 
-@Composable
-fun EventCategory(modifier: Modifier, contentColor: Color, titleColor: Color) {
-  Column(modifier = modifier, verticalArrangement = Arrangement.SpaceBetween) {
-    Text(
-        text = headerCategory,
-        style = subTitleTextStyle,
-        color = titleColor,
-        modifier = Modifier.testTag("categoryTitle"))
-    Text(
-        text = contentCategory,
-        style = standardTextStyle,
-        color = contentColor,
-        modifier = Modifier.testTag("categoryContent"))
-  }
-}
+  viewModel.getEventData()
+  val eventUiState = viewModel.uiState.collectAsState().value
 
-@Composable
-fun EventTime(modifier: Modifier, contentColor: Color, titleColor: Color) {
-  Column(modifier = modifier) {
-    Text(
-        text = headerTime,
-        style = subTitleTextStyle,
-        color = titleColor,
-        modifier = Modifier.testTag("timeTitle"))
-    Text(
-        text = "start $contentTime",
-        style = standardTextStyle,
-        color = contentColor,
-        modifier = Modifier.testTag("timeStartContent"))
-    Text(
-        text = "end $contentTime",
-        style = standardTextStyle,
-        color = contentColor,
-        modifier = Modifier.testTag("timeEndContent"))
-  }
-}
-
-@Composable
-fun EventDetails(navigationActions: NavigationActions) {
-
-  val fieldTitleColor = MaterialTheme.colorScheme.onSurfaceVariant
-  val fieldContentColor = MaterialTheme.colorScheme.onSurface
+  val componentStyle =
+      EventComponentsStyle(
```

Comment:
Since I separted the "description "distance" "date & time" etc.. in different composable function I wanted to avoid
- having a tons of function parameters
- to hard code the color scheme in each component when they need to be harmonized

Reply:
Ok, make sure this is in line with what the rest of the app looks like.

Reply:
Is the color the same as the rest of the app? I don't think we have implemented specific colors like you have (we are currently just using default colors). If that is the case, then maybe take this out of this PR for now so it can be uniform for M2.